### PR TITLE
fix: Disable deprecated xla-arguments

### PR DESCRIPTION
**Pull request by ko3n1g**
XLA upstream removed
`xla_gpu_enable_async_all_gather` and
`xla_gpu_enable_async_reduce_scatter`
@URL: https://github.com/openxla/xla/commit/4298a065f7b8381d690a95fe6d437cff037b3ff4

### CODE REVIEW

File: rosetta/docs/GPU_performance.md [hunk 1]
```diff
@@ -42,17 +42,8 @@ To enable latency hiding optimizations with XLA, turn on the following flag:
 
 To enable asynchronous communication for all collectives, the following is recommended, and is set by default in XLA :
 
-- --xla_gpu_enable_async_collectives=true
 - --xla_gpu_enable_highest_priority_async_stream=true
 
-For more fine-grained control over which collectives should be asynchronous or not, please use:
```
Comment (olupton) [inline rosetta/docs/GPU_performance.md:48]: I think we should keep this sentence, but replace the bullets below with `--xla_gpu_disable_async_collectives=allreduce,allgather,reducescatter,collectivebroadcast,alltoall,collectivepermute` or similar.

File: rosetta/docs/PGLE.md [hunk 2]
```diff
@@ -8,9 +8,8 @@ The workflow to use the Profile Guided Latency Estimator with Nsight Systems (Ns
 1. **Profile Run**: We need to run the workload once, with the async collectives and latency hiding scheduler disabled. To be specific we need to turn the following three XLA flags off. All the remaining flags should be unchanged.
 
 ```
-export XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=false
---xla_gpu_enable_async_all_gather=false
---xla_gpu_enable_async_reduce_scatter=false"
+export XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=false"
```
Comment (olupton) [inline rosetta/docs/PGLE.md:11]: I think here we need `--xla_gpu_disable_async_collectives=allreduce,allgather,reducescatter,collectivebroadcast,alltoall,collectivepermute` to disable them all.

File: rosetta/rosetta/projects/diffusion/common/set_gpu_xla_flags.sh [hunk 3]
```diff
@@ -1 +1 @@
-export XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=false --xla_gpu_enable_async_all_gather=false --xla_gpu_enable_async_reduce_scatter=false --xla_gpu_enable_triton_gemm=false --xla_gpu_cuda_graph_level=0 --xla_gpu_enable_triton_softmax_fusion=false --xla_gpu_enable_async_all_reduce=false ${XLA_FLAGS}"
\ No newline at end of file
+export XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=false --xla_gpu_enable_triton_gemm=false --xla_gpu_cuda_graph_level=0 --xla_gpu_enable_triton_softmax_fusion=false ${XLA_FLAGS}"
```
Comment (olupton) [inline rosetta/rosetta/projects/diffusion/common/set_gpu_xla_flags.sh:1]: I don't know the origins of this script. But async collectives were being disabled here, so it should be `--xla_gpu_disable_async_collectives=allreduce,allgather,reducescatter,collectivebroadcast,alltoall,collectivepermute`.

**Review by olupton — CHANGES_REQUESTED** [review]

File: rosetta/docs/PGLE.md (hunk 2 above)
Comment (ko3n1g) [reply rosetta/docs/PGLE.md]: I assume I should add these on-top instead of replacing the existing args, right?

Comment (olupton) [reply rosetta/docs/PGLE.md]: I believe that `--xla_gpu_disable_async_collectives=allgather,reducescatter` is the new way of spelling `--xla_gpu_enable_async_all_gather=false --xla_gpu_enable_async_reduce_scatter=false`.

File: rosetta/rosetta/projects/diffusion/common/set_gpu_xla_flags.sh [hunk 4]
```diff
@@ -1 +1 @@
-export XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=false --xla_gpu_enable_async_all_gather=false --xla_gpu_enable_async_reduce_scatter=false --xla_gpu_enable_triton_gemm=false --xla_gpu_cuda_graph_level=0 --xla_gpu_enable_triton_softmax_fusion=false --xla_gpu_enable_async_all_reduce=false ${XLA_FLAGS}"
\ No newline at end of file
+export XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=false --xla_gpu_enable_triton_gemm=false --xla_gpu_cuda_graph_level=0 --xla_gpu_enable_triton_softmax_fusion=false --xla_gpu_disable_async_collectives=allreduce,allgather,reducescatter,collectivebroadcast,alltoall,collectivepermute${XLA_FLAGS}"
```
Comment (olupton) [inline rosetta/rosetta/projects/diffusion/common/set_gpu_xla_flags.sh:1]: Missing space before `${XLA_FLAGS}`

Comment (ko3n1g) [reply rosetta/rosetta/projects/diffusion/common/set_gpu_xla_flags.sh]: thanks, good catch!

**Review by olupton — APPROVED** [review]
Looks good. We may like to follow up later and see if any more flags can be dropped, but I think this is a good translation to the new flag spellings.

**Pull request merged**